pkg/services/repobuilder.go

```
@@ -128,23 +128,23 @@ func (rb *RepoBuilder) BuildUpdateRepo(ut *models.UpdateTransaction) (*models.Up
 		return nil, err
 	}
 
-	if update.Repo == nil {
-		//  Check for the existence of a Repo that already has this commit and don't duplicate
-		var repo *models.Repo
-		repo, err = rb.repoService.GetRepoByCommitID(update.CommitID)
-		if err == nil {
-			update.Repo = repo
-		} else {
-			if err.Error() != "record not found" {
-				log.Errorf("updateFromHTTP::GetRepoByCommitID::repo: %#v, %#v", repo, err)
-			} else {
-				log.Infof("Old Repo not found in database for CommitID, creating new one: %d", update.CommitID)
-				update.Repo = &models.Repo{
-					Commit: update.Commit,
-				}
-			}
-		}
-	}
+	// if update.Repo == nil {
```

Just delete it? We can recover over git history if needed

---

Or better: if update.Repo is nil here, we might want to return an error in the whole method, just to avoid a nil pointer that will be hard to track